### DISCUSSION TITLE
vast.ai notebook

### DISCUSSION THREAD

**Discussion by Alex3191** [acids-ircam/RAVE#147] [Q&A] [answered]
hi, has anyone modified the colab-notebook for usage on vast.ai? i am trying to migrate my work away from colab because of the recent modification in price structure as it has become quite expensive. in comparison the cloud gpu service vast.ai seems much cheaper. however i am struggling to modify the kindly provided jupyter-notebook for colab for a usage in vast.ai. i have difficulties with installing conda related libraries / the file structure on vast.ai. 

so far i'm trying to modify the install section accordingly. the notebook was opened from the first workspace directory of my opened instance on vast.ai ( cd /workspace).

anyone can help?

%cd /content/drive/MyDrive
!git clone --depth 1 https://github.com/acids-ircam/RAVE.git RAVE_COLLAB
%cd /content/drive/MyDrive/RAVE_COLLAB/
!curl -L https://repo.anaconda.com/miniconda/Miniconda3-py39_4.12.0-Linux-x86_64.sh -o miniconda.sh
!chmod +x miniconda.sh
!sh miniconda.sh -b -p /content/miniconda
!/content/miniconda/bin/pip install -r requirements.txt
!/content/miniconda/bin/pip install --upgrade ipython ipykernel

**Comment by Alex3191** [comment]
the solution would be the following, however running the various scripts (resample, training) results in sndfile missing, and installing sndfile results in an error. what am i missing here, is it the python version?

!git clone --depth 1 https://github.com/acids-ircam/RAVE.git RAVE
%cd /workspace/RAVE
!curl -L https://repo.anaconda.com/miniconda/Miniconda3-py39_4.12.0-Linux-x86_64.sh -o miniconda.sh
!chmod +x miniconda.sh
!sh miniconda.sh -b -p /workspace/miniconda
!/workspace/miniconda/bin/pip install -r requirements.txt
!/workspace/miniconda/bin/pip install --upgrade ipython ipykernel

**Reply by Alex3191** [reply]
solved by running on the machine via terminal

sudo apt-get update
sudo apt-get install libsndfile1

also need to run pip install tensorflow in the notebook

HOWEVER now the scripts dont work properly, resample doesnt output a file and training (adjusted the path) gives me:
Traceback (most recent call last):
File "/workspace/RAVE/train_rave.py", line 156, in
nepoch = args.VAL_EVERY // len(train)
ZeroDivisionError: integer division or modulo by zero

**Comment by npnp** [comment, marked as answer]
Hi,
perhaps there are some leftovers from previous experiments with notebook etc?
Maybe if you try again from zero?

I've managed today to run the training on vast.ai instance over ssh with a 3090 (4090s didn't work because of some bug in pytorch cuda 11.7 that was fixed in 11.8 but is not yet offered by pytorch conda install).

My steps were:
1. Create instance with pytorch latest image, ssh into it
4. Do the steps from rave install
3. update pytorch through conda:
`conda install pytorch torchvision torchaudio pytorch-cuda=11.7 -c pytorch -c nvidia`
4. install ffmpeg (for resample):
`apt-get install ffmpeg`

At this point training and resample worked for me

**Reply by Alex3191** [reply]
hi, thank you so much! :) i was finally able to run the training finally after following your instructions (havent tried out resample but shouldnt be a problem i guess). i will soon try run the whole notebook again as i now know that gpu type, cuda version etc all matter.

**Reply by Alex3191** [reply]
hi again, 
i wanted to run the new v2, however the preprocessing doesnt really work / it gets stuck.

(also libdsndfile was missing which i installed via: conda install -c conda-forge libsndfile )

so when i run the preprocess script, it gets stuck as in the pic attached. running it in lazy mode gets me a bit further as it seems to at least produce the metadata file, however it also doesnt really seem to process anything and i still cannot start the training.

any idea?

![Bildschirmfoto 2023-02-09 um 17 42 26](https://user-images.githubusercontent.com/120024841/218163983-7048e234-c62e-41e3-8a24-bde76d376a02.png)
